packages/frontend/src/components/Header.jsx

```diff
@@ -22,19 +23,21 @@ export default observer(() => {
   }, [])
   const [searchInput, setSearchInput] = useState('')
 
-  // currently search doesn't check for any matching attester or user, just goes to 'user/searchInput'
-
-  // const [goToPage, setGoToPage] = useState('')
-  // const handleGo = () => {
-  //     if (userIds.some(e => e.commitment === searchInput)) {
-  //         setGoToPage(`user/${searchInput}`)
-  //     } else if (attesterIds.includes(searchInput)) {
-  //         setGoToPage(`attester/${searchInput}`)
-  //     } else {
-  //         setGoToPage('404')
-  //     }
-  //     setSearchInput('')
-  // }
+  const search = () => {
+    if (!/^(0x)?[a-fA-F0-9]*$/.test(searchInput)) {
+      // invalid input, only accept hexadecimal
+      // TODO: highlight the field red or something
+      return
+    }
+    const val = BigInt(`0x${searchInput.replace('0x', '')}`)
+    if (val > BigInt(2) ** BigInt(160)) {
+      // it's an epoch key
+      navigate(`/user/${searchInput})`)
+    } else {
+      // otherwise treat it as an attester id
+      navigate(`/attester/${searchInput})`)
+    }
```

Comment:
`UserPage` displays signup info for an `identity commitment`, not its associated epoch keys, so I'm confused by this.. why would you be able to search for epoch keys and be able to get info about the id that owns them?

Reply:
Ah, we'll need to be able to search for either an identity commitment or an epoch key. I think identity commitments are less useful though because they're only seen once.